### PR TITLE
Add cli param to disable default ignores

### PR DESCRIPTION
**Pull request by velrest**
This is a second attempt for #659 

Add flag so the by default ignored files
`['**/dist/**', '**/tmp/**', '**/node_modules/**']` and `.gitignore`
can be toggled. This is useful if for instance your editor lint plugin
(vim/ale) copies the file to be linted into /tmp . This prevents the
template from being linted as '**/tmp/**' applies.

Also add `--help` command to list available options.

### CODE REVIEW

**Comment by rwjblue** [comment]
> Also add --help command to list available options.

Very nice!

**Comment by rwjblue** [comment]
Looks like CI failures are due to linting issues:

```
$ eslint .
/home/travis/build/ember-template-lint/ember-template-lint/bin/ember-template-lint.js
  104:22  error  'toProcess' is defined but never used             no-unused-vars
  115:22  error  'toProcess' is defined but never used             no-unused-vars
  121:22  error  'toProcess' is defined but never used             no-unused-vars
  127:22  error  'toProcess' is defined but never used             no-unused-vars
  133:22  error  'toProcess' is defined but never used             no-unused-vars
  161:26  error  Strings must use singlequote                      quotes
  162:13  error  Don't use process.exit(); throw an error instead  no-process-exit
  163:13  error  Unreachable code                                  no-unreachable
✖ 8 problems (8 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by rwjblue** [comment]
What do you think about changing the option to `--ignore-patterns` and taking a comma separated list of patterns to ignore, and if specified `--ignore-patterns` would then replace the default ignore patterns?

File: README.md [hunk 1]
```diff
@@ -43,13 +43,14 @@ var results = linter.verify({ source: template, moduleId: 'template.hbs' });
...
-* `source` - The source that caused the error.
-* `fix` - An object describing how to fix the error.
+
+- `rule` - The name of the rule that triggered this warning/error.
+- `message` - The message that should be output.
+- `line` - The line on which the error occurred.
+- `column` - The column on which the error occurred.
+- `moduleId` - The module path for the file containing the error.
+- `source` - The source that caused the error.
+- `fix` - An object describing how to fix the error.
```
Comment (Turbo87) [inline README.md]: why did you change the syntax here?

File: README.md [hunk 2]
```diff
@@ -80,7 +81,10 @@ Example usage:
 ./node_modules/.bin/ember-template-lint --filename app/templates/application.hbs < app/templates/application.hbs
 
 # print list of formated rules for use with `pending` in config file
-./node_modules/.bin/ember-template-lint app/templates/application.hbs --print-pending
+./node_modules/.bin/ember-template-lint app/templaes/application.hbs --print-pending
```
Comment (Turbo87) [inline README.md]: there is a `t` missing in `templaes`

**Comment by dcyriller** [comment]
@velrest you might be interested by https://github.com/dense-analysis/ale/pull/2622 😉. I have to finish it.

File: README.md (hunk 1 above)
Comment (velrest) [reply README.md]: Sorry prettier reformatted the whole file

**Comment by velrest** [comment]
> What do you think about changing the option to `--ignore-patterns` and taking a comma separated list of patterns to ignore, and if specified `--ignore-patterns` would then replace the default ignore patterns?

@rwjblue Im not so sure if `ember-template-lint --ignore-patterns` would be a good pattern for disabling the default ignore patterns. Do they need to be hard coded in the executable? Since we already respect the `.gitignore` (which probably has these ignore patterns already configured) what about removing the hard coded patterns entirely or is this not possible?  I agree with a cli option to set custom ignore-patterns though. Maybe even via `.template-lint.rc`?

**Comment by dcyriller** [comment]
@velrest to understand why there are hard coded ignored patterns, you can a have a look to this issue: https://github.com/ember-template-lint/ember-template-lint/issues/536
TL;DR: globby doesn’t always parse `.gitignore` as expected, especially in monorepos where there is only one top-level `.gitignore`.

Also: do you think you could extract the —help option from your PR? It looks like a feature less complex than this PR’s. We could review and merge it while finding a good solution for this pattern issue.

**Comment by rwjblue** [comment]
Ya, totally agree. The parsing and `--help` updates here are quite nice, and I'd love to land them ASAP...

**Comment by velrest** [comment]
Sorry for the long wait.

I opened ember-template-lint/ember-template-lint#1074 for the --help and cli options parsing.

**Comment by rwjblue** [comment]
Thanks again for all of your work here @velrest! @kangax picked this up and finished it off over in #1200. Closing this now...
